### PR TITLE
New package: xidlehook-0.10.0

### PR DESCRIPTION
**Pull request by MOIS3Y**
This PR is an updated version of https://github.com/void-linux/void-packages/pull/27774 since the original is closed. The requested changes have also been added.
There's also an associated package request for this PR: https://github.com/void-linux/void-packages/issues/27422

#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)
- I built this PR locally for these architectures:
    - armv6l

### CODE REVIEW

File: srcpkgs/xidlehook/template [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+build_wrksrc="xidlehook-daemon"
+build_style=cargo
+configure_args="$(vopt_if audio '--features pulse' '--no-default-features')"
+hostmakedepends="pkg-config python3"
+makedepends="libxcb-devel libXScrnSaver-devel $(vopt_if audio pulseaudio-devel)"
+short_desc="Xautolock rewrite in Rust, with a few extra features"
+maintainer="MOIS3Y <stepan@zhukovsky.me>"
+license="MIT"
+homepage="https://gitlab.com/jD91mZM2/xidlehook"
+distfiles="${homepage}/-/archive/${version}/${pkgname}-${version}.tar.gz"
```
Comment (Duncaen) [inline srcpkgs/xidlehook/template:14]: Could you remove the variables `homepage` and `pkgname` here, they are not variable.

File: srcpkgs/xidlehook/template [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+checksum=66751b78c5174c0e430d9becd749771d593f388db348514d8c3256670dc77cd8
+build_options_default="audio"
+desc_option_audio="Don't invoke the timer when any audio is playing (Only works with Pulseaudio)"
```
Comment (Duncaen) [inline srcpkgs/xidlehook/template:17]: Randomly named build options are hardly useful, this should probably just use the `pulseaudio` build option.

File: srcpkgs/xidlehook/template (hunk 1 above)
Comment (MOIS3Y) [reply srcpkgs/xidlehook/template]: Done

File: srcpkgs/xidlehook/template (hunk 2 above)
Comment (MOIS3Y) [reply srcpkgs/xidlehook/template]: Done too.
After the changes, I re-ran the build and check with the linter
```sh
./xbps-src pkg xidlehook
./xbps-src -a armv6l -Q pkg xidlehook
xlint srcpkgs/xidlehook/template
```

**Pull request closed without merging**